### PR TITLE
Add `get_platform_shortname`; update BFD platform-specific settings

### PR DESCRIPTION
**Pull request by chrisvanheuveln**
##### SUMMARY
see title

### CODE REVIEW

File: lib/ansible/module_utils/network/nxos/nxos.py [hunk 1]
```diff
@@ -726,33 +726,46 @@ def feature_enable(self):
...
-        info = get_capabilities(self._module).get('device_info')
-        os_platform = info.get('network_os_platform')
-        plat = None
-
-        # Supported Platform IDs (N35 N3K N5K N6K N7K N9K)
-        if not os_platform:
+    def get_platform_shortname(self):
+        """Query device for platform type, normalize to a shortname/nickname.
+        Returns platform shortname (e.g. 'N3K-3058P' returns 'N3K') or None.
+        """
```
Comment (glennmatthews) [inline lib/ansible/module_utils/network/nxos/nxos.py:732]: Do y'all have any UT automation (or doctests) for this module? This function could really use a test matrix of sample platform outputs and the expected return value.

Comment (chrisvanheuveln) [reply lib/ansible/module_utils/network/nxos/nxos.py:732]: Not yet. I still have to write the UT's. I'll do that next and open a separate PR.

File: lib/ansible/module_utils/network/nxos/nxos.py [hunk 2]
```diff
@@ -726,33 +726,46 @@ def feature_enable(self):
...
+        # TBD: add this method logic to get_capabilities() after those methods
+        #      are made consistent across transports
+        platform_info = self.execute_show_command('show inventory', 'json')
```
Comment (mikewiebe) [inline lib/ansible/module_utils/network/nxos/nxos.py:735]: Looks like platform_info could in theory contain an error, not just `None`.  

https://github.com/chrisvanheuveln/ansible/blob/8344ec6c52aab5893b27ef85f09a522a479a227b/lib/ansible/module_utils/network/nxos/nxos.py#L801

Comment (chrisvanheuveln) [reply lib/ansible/module_utils/network/nxos/nxos.py:735]: Hmmm, only if `'show inventory'` fails to process on the device, in which case there are much bigger problems going on.  But I'll add a check for dict just to be safe.

**Review by mikewiebe — APPROVED** [review]

**Pull request merged**